Move native MHA code out of PyTorch core

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72944

Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

swolchok committed:
Move native MHA code out of PyTorch core

Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

ezyang reviewed (approved):
not sure what the high level resolution was (missed the meeting), but assuming it's this, diff looks good

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]

swolchok committed:
Update on "Move native MHA code out of PyTorch core"


Doesn't make sense to develop it in core right now.

Differential Revision: [D34283104](https://our.internmc.facebook.com/intern/diff/D34283104/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34283104/)!

[ghstack-poisoned]
